### PR TITLE
Adding initial support for MOOSE classes snippets

### PR DESCRIPTION
**Pull request by andrsd**

### CODE REVIEW

**Comment by dschwen** [comment]
:+1:
I'll give this a spin.
There is also https://github.com/atom/snippets/issues/170 which - when implemented - could be used for the include guards.

**Comment by andrsd** [comment]
> There is also atom/snippets#170 which - when implemented - could be used for teh include guards.

That would be great!

**Comment by andrsd** [comment]
Note that for PPS, UO and VectorPPS you need to also provide the base class, i.e. Element, Side, General, Nodal, InternalSide-XYZ.

File: snippets/action.cson [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public:
+      ${2}(const InputParameters & parameters);
+
+      virtual void act();
+    };
+
+    #endif // ${1}_H
+    """
+
+'.source.cpp':
```
Comment (dschwen) [inline snippets/action.cson:31]: an empty `.C` file defaults to `.source.c` for me. Did you do something to have it set to `.source.cpp`?

Comment (andrsd) [inline snippets/action.cson:31]: I think I changed it locally, so that language-c is loading C++ grammar instead of C one. I forgot about that ;-)

Comment (dschwen) [inline snippets/action.cson:31]: Ok, I'll fix this up like so: 5a7f224 and merge if that is fine with you

Comment (andrsd) [inline snippets/action.cson:31]: That's fine.

**Comment by dschwen** [comment]
Ok, I added fe0d2caef0ea and published this a v0.17.0
Thanks!

**Pull request merged**